src/mig/modules/pkgprint/pkgprint.go

```diff
@@ -0,0 +1,347 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Aaron Meihm ameihm@mozilla.com [:alm]
+package pkgprint
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mig/modules"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+)
+
+func init() {
+	modules.Register("pkgprint", func() interface{} {
+		return new(Runner)
+	})
+}
+
+type FPResult struct {
+	Name    string       `json:"name"`
+	Matches []MatchGroup `json:"matches"`
+}
+
+type MatchGroup struct {
+	Root    string       `json:"root"`
+	Entries []MatchEntry `json:"entries"`
+}
+
+type MatchEntry struct {
+	Match string `json:"match"`
+}
+
+func executeTemplate(tname string, depth int, root string) (FPResult, error) {
+	var (
+		res FPResult
+		err error
+	)
+	fp := getTemplateFingerprint(tname)
+	if fp == nil {
+		return res, fmt.Errorf("invalid template name specified")
+	}
+	res.Name = tname
+	s := NewSimpleFileLocator()
+	s.maxDepth = depth
+	if root != "/" {
+		s.root = root
+	}
+
+	// If the fingerprint has a forced root, override the root that has
+	// been specified.
+	if fp.forceRoot != "" {
+		s.root = fp.forceRoot
+	}
+
+	s.Locate(fp.filename, fp.isRegexp)
+	for _, x := range s.matches {
+		// If a path filter exists and does not match the file, ignore
+		// it.
+		if fp.pathFilter != "" {
+			var flag bool
+			flag, err = regexp.MatchString(fp.pathFilter, x)
+			if err != nil || !flag {
+				continue
+			}
+		}
+
+		// If an error occurs here, just ignore it and keep going.
+		m, _ := FileContentCheck(x, fp.contentMatch)
+		if m == nil || len(m) == 0 {
+			continue
+		}
+		for _, i := range m {
+			// The module requires a substring match in the
+			// template, so we only process entries that have
+			// resulted in substrings being returned.
+			if len(i) < 2 {
+				continue
+			}
+			newmatch := MatchGroup{}
+			newmatch.Root = x
+			for j := 1; j < len(i); j++ {
+				nme := MatchEntry{}
+				nme.Match, err = fp.transform(i[j])
+				if err != nil {
+					continue
+				}
+				newmatch.Entries = append(newmatch.Entries, nme)
+			}
+			res.Matches = append(res.Matches, newmatch)
+		}
+	}
+	return res, nil
+}
+
+// Given a file, identify any lines in the file that match the supplied
+// regular expression. Will return a slice of string slices. If the
+// supplied regular expression contains substring matches, each string
+// slice will contain the full line matched, and any substrings, otherwise
+// the slice will only contain the line matched.
+func FileContentCheck(path string, regex string) ([][]string, error) {
```

Comment:
Indeed, this does replicate functionality from the `file` module. Have you look into calling that module yet?